src/main.rs

```diff
@@ -91,6 +91,17 @@ fn main() {
             .open(tbf_path.clone())
             .unwrap();
 
+        // Adding padding to the end of cortex-m apps. Check for a cortex-m app
+        // because the start of the elf filename will start with "cortex".
+        //
+        // RISC-V apps do not need to be sized to power of two.
+        let add_trailing_padding = elf_path
+            .file_name()
+            .unwrap_or(std::ffi::OsStr::new(""))
+            .to_str()
+            .unwrap_or("")
+            .starts_with("cortex");
```

Comment:
Doing this based on filename feels pretty precarious, isn't the architecture metadata in the elf file that this is already parsing?

```
$ file cortex-m0/cortex-m0.elf
cortex-m0/cortex-m0.elf: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, with debug_info, not stripped
```

Even the `file` utility knows this is for ARM..